### PR TITLE
Add new issue that demonstrates how to use PartitionedRateLimiter

### PR DESCRIPTION
**Pull request by martintmk**
### Details on the issue fix or feature implementation

Contributes to #1365 

### Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -9,6 +9,8 @@
     <PackageVersion Include="BenchmarkDotNet" Version="0.13.5" />
     <PackageVersion Include="FluentAssertions" Version="6.11.0" />
     <PackageVersion Include="GitHubActionsTestLogger" Version="2.3.2" />
+    <PackageVersion Include="Microsoft.AspNetCore.Http" Version="2.2.2" />
+    <PackageVersion Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
```
Comment (martincostello) [inline Directory.Packages.props:13]: These packages are deprecated - we should add [a framework reference](https://learn.microsoft.com/aspnet/core/fundamentals/target-aspnetcore#use-the-aspnet-core-shared-framework) to the tests instead.

File: test/Polly.Extensions.Tests/Polly.Extensions.Tests.csproj [hunk 2]
```diff
@@ -16,7 +16,10 @@
   <ItemGroup>
     <Using Include="Polly.TestUtils" />
     <ProjectReference Include="..\..\src\Polly.Extensions\Polly.Extensions.csproj" />
+    <ProjectReference Include="..\..\src\Polly.RateLimiting\Polly.RateLimiting.csproj" />
     <ProjectReference Include="..\Polly.TestUtils\Polly.TestUtils.csproj" />
+    <PackageReference Include="Microsoft.AspNetCore.Http" />
+    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" />
     <PackageReference Include="Microsoft.Extensions.Configuration" />
     <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" />
```
Comment (martincostello) [inline test/Polly.Extensions.Tests/Polly.Extensions.Tests.csproj:24]:
```suggestion
    <ProjectReference Include="..\..\src\Polly.Extensions\Polly.Extensions.csproj" />
    <ProjectReference Include="..\..\src\Polly.RateLimiting\Polly.RateLimiting.csproj" />
    <FrameworkReference Include="Microsoft.AspNetCore.App" />
    <PackageReference Include="Microsoft.Extensions.Configuration" />
    <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" />
```

File: Directory.Packages.props (hunk 1 above)
Comment (martintmk) [reply Directory.Packages.props:13]: Do I need to actually use `IHttpContextAccessor`? Maybe I can demonstrate it just on the `ResilienceContext`?

Comment (martincostello) [reply Directory.Packages.props:13]: I don't mind - the HTTP context way might be more familiar to people though.

Comment (martintmk) [reply Directory.Packages.props:13]: I'll add the comment about it in the test.

File: test/Polly.Extensions.Tests/Polly.Extensions.Tests.csproj (hunk 2 above)
Comment (martintmk) [reply test/Polly.Extensions.Tests/Polly.Extensions.Tests.csproj]: Not relevant as no new dependencies were added.

**Review by martincostello — APPROVED** [review]

**Pull request merged**